Mathlib/Data/IsROrC/Basic.lean

```diff
@@ -56,379 +56,365 @@ open ComplexConjugate
+  ofReal_re_ax : ∀ r : ℝ, re (𝓚 r) = r
+  ofReal_im_ax : ∀ r : ℝ, im (𝓚 r) = 0
   mul_re_ax : ∀ z w : K, re (z * w) = re z * re w - im z * im w
   mul_im_ax : ∀ z w : K, im (z * w) = re z * im w + im z * re w
   conj_re_ax : ∀ z : K, re (conj z) = re z
   conj_im_ax : ∀ z : K, im (conj z) = -im z
-  conj_i_ax : conj I = -I
+  conj_I_ax : conj I = -I
   norm_sq_eq_def_ax : ∀ z : K, ‖z‖ ^ 2 = re z * re z + im z * im z
-  mul_im_i_ax : ∀ z : K, im z * im I = im z
+  mul_im_I_ax : ∀ z : K, im z * im I = im z
 #align is_R_or_C IsROrC
 
 end
 
-/- failed to parenthesize: unknown constant 'Lean.Meta._root_.Lean.Parser.Command.registerSimpAttr'
-[PrettyPrinter.parenthesize.input] (Lean.Meta._root_.Lean.Parser.Command.registerSimpAttr
-     [(Command.docComment "/--" "Simp attribute for lemmas about `is_R_or_C` -/")]
-     "register_simp_attr"
-     `is_R_or_C_simps)-/-- failed to format: unknown constant 'Lean.Meta._root_.Lean.Parser.Command.registerSimpAttr'
-/-- Simp attribute for lemmas about `is_R_or_C` -/ register_simp_attr is_R_or_C_simps
-
 variable {K E : Type _} [IsROrC K]
 
 namespace IsROrC
 
 open ComplexConjugate
 
+/-- Coercion from `ℝ` to an `IsROrC` field. -/
+@[coe] abbrev ofReal : ℝ → K := Algebra.cast
+
 /- The priority must be set at 900 to ensure that coercions are tried in the right order.
 See Note [coercion into rings], or `data/nat/cast.lean` for more details. -/
-noncomputable instance (priority := 900) algebraMapCoe : CoeTC ℝ K :=
-  ⟨algebraMap ℝ K⟩
+noncomputable instance (priority := 900) algebraMapCoe : CoeTC ℝ K := ⟨ofReal⟩
 #align is_R_or_C.algebra_map_coe IsROrC.algebraMapCoe
 
-theorem of_real_alg (x : ℝ) : (x : K) = x • (1 : K) :=
-  Algebra.algebraMap_eq_smul_one x
-#align is_R_or_C.of_real_alg IsROrC.of_real_alg
+theorem ofReal_alg (x : ℝ) : (x : K) = x • (1 : K) := Algebra.algebraMap_eq_smul_one x
+#align is_R_or_C.of_real_alg IsROrC.ofReal_alg
 
-theorem real_smul_eq_coe_mul (r : ℝ) (z : K) : r • z = (r : K) * z :=
-  Algebra.smul_def r z
+theorem real_smul_eq_coe_mul (r : ℝ) (z : K) : r • z = (r : K) * z := Algebra.smul_def r z
 #align is_R_or_C.real_smul_eq_coe_mul IsROrC.real_smul_eq_coe_mul
 
 theorem real_smul_eq_coe_smul [AddCommGroup E] [Module K E] [Module ℝ E] [IsScalarTower ℝ K E]
-    (r : ℝ) (x : E) : r • x = (r : K) • x := by rw [IsROrC.of_real_alg, smul_one_smul]
+    (r : ℝ) (x : E) : r • x = (r : K) • x := by rw [IsROrC.ofReal_alg, smul_one_smul]
 #align is_R_or_C.real_smul_eq_coe_smul IsROrC.real_smul_eq_coe_smul
 
-theorem algebraMap_eq_of_real : ⇑(algebraMap ℝ K) = coe :=
-  rfl
-#align is_R_or_C.algebra_map_eq_of_real IsROrC.algebraMap_eq_of_real
+theorem algebraMap_eq_ofReal : ⇑(algebraMap ℝ K) = ofReal := rfl
+#align is_R_or_C.algebra_map_eq_of_real IsROrC.algebraMap_eq_ofReal
 
-@[simp, is_R_or_C_simps]
-theorem re_add_im (z : K) : (re z : K) + im z * i = z :=
-  IsROrC.re_add_im_ax z
+@[simp, isROrC_simps]
+theorem re_add_im (z : K) : (re z : K) + im z * I = z := IsROrC.re_add_im_ax z
```

Comment:
Moving the proofs to the previous line relative to the mathport output makes this PR harder to review and makes it harder to forward-fort any changes that may still happen in mathlib3 in the next few months. Is there a compelling reason for it?

Reply:
Done. Any ideas why linter fails in seemingly unrelated files?